### PR TITLE
Common types

### PR DESCRIPTION
**Pull request by joelrebel**
#### What does this PR do

1. Switch to common.Device type and constants from the common pacakge.

The common.Device type is provided by https://github.com/bmc-toolbox/common
the ironlib device and components are now of the common.Device underlying type.

The changes included updates all Device, Component types and Slug
constants to work with the common types and constants.

Any tooling that now builds on top ironlib (and bmclib)
can work with a consistent (common) set of Device, Component, Firmware types and
Slug constants. This means theres less work needed to be done to transform
between types from both the libraries, and compare slugs/constants.

2. Update golang ci-linter to 1.46.2

### CODE REVIEW

File: model/device.go [hunk 1]
```diff
@@ -13,182 +15,14 @@ func NewHardware(d *Device) *Hardware {
...
-		GPUs:               []*GPU{},
-		CPUs:               []*CPU{},
-		Memory:             []*Memory{},
-		Drives:             []*Drive{},
-		StorageControllers: []*StorageController{},
-	}
+	return &Device{common.NewDevice(), nil}
 }
 
+// Device is an ironlib device object which extends the common.Device
```
Comment (diogomatsubara) [inline model/device.go:23]: It doesn't seem that we're extending the ironlib device object anymore. Shouldn't we get rid of the `ironlib.Device` entirely and just use the bmclib one directly?

File: utils/dmidecode.go [hunk 2]
```diff
@@ -25,9 +26,24 @@ func NewDmidecode() (d *Dmidecode, err error) {
 	return &Dmidecode{dmi: dmi}, err
 }
 
-// Returns a fake dmidecode instance for tests
-func NewFakeDmidecode() *Dmidecode {
-	return &Dmidecode{}
+// InitFakeDmidecode Returns a fake dmidecode instance for tests
+// given test data file returns a Dmidecode with the test dmidecode output loaded
```
Comment (diogomatsubara) [inline utils/dmidecode.go:30]: I'd suggest: 

```
// InitFakeDmidecode returns a fake Dmidecode instance loaded from the dmidecode output in testFile
```

File: utils/mlxup.go [hunk 3]
```diff
@@ -45,14 +45,14 @@ func NewMlxupCmd(trace bool) *Mlxup {
 	return &Mlxup{Executor: e}
 }
 
-// Collect returns a slice of mellanox components as *model.NIC's
-func (m *Mlxup) NICs(ctx context.Context) ([]*model.NIC, error) {
+// Collect returns a slice of mellanox components as *common.NIC's
```
Comment (diogomatsubara) [inline utils/mlxup.go:48]: `NICs returns a slice of mellanox...`

File: utils/msecli.go [hunk 4]
```diff
@@ -42,23 +43,26 @@ func NewMsecli(trace bool) *Msecli {
 }
 
 // Components returns a slice of drive components identified
```
Comment (diogomatsubara) [inline utils/msecli.go:45]: `Drives returns a slice of drive...`

File: utils/nvme.go [hunk 5]
```diff
@@ -40,9 +40,9 @@ func NewNvmeCmd(trace bool) *Nvme {
 	return &Nvme{Executor: e}
 }
 
-// Executes nvme list, parses the output and returns a slice of *model.Drive
-func (n *Nvme) Drives(ctx context.Context) ([]*model.Drive, error) {
-	drives := make([]*model.Drive, 0)
+// Executes nvme list, parses the output and returns a slice of *common.Drive
+func (n *Nvme) Drives(ctx context.Context) ([]*common.Drive, error) {
```
Comment (diogomatsubara) [inline utils/nvme.go:44]: `Drives returns a slice of *model.Drive from the output of nvme list`

File: utils/smc_ipmicfg.go [hunk 6]
```diff
@@ -49,58 +49,67 @@ func NewFakeIpmicfg(r io.Reader) *Ipmicfg {
...
+				Installed: summary.BIOSVersion,
+				Metadata: map[string]string{
+					"build_date": summary.BIOSBuildDate,
+				},
 			},
 		},
 	}, nil
 }
 
 // CPLD returns a SMC CPLD component
```
Comment (diogomatsubara) [inline utils/smc_ipmicfg.go:94]: `CPLDs returns a slice of SMC CPLD components`

File: utils/storecli.go [hunk 7]
```diff
@@ -69,8 +69,8 @@ func NewFakeStoreCLI(r io.Reader) (*StoreCLI, error) {
 }
 
 // Executes nvme list, parses the output and returns a slice of model.StorageControllers
```
Comment (diogomatsubara) [inline utils/storecli.go:71]: `StorageControllers returns a slice of model.StorageControllers from the output of nvme list`

**Review by diogomatsubara — CHANGES_REQUESTED** [review]
Thanks for the work on this.

I made some comments mainly on documentation that needs update after changing the methods signatures.

Also it seems we can get rid of the `model.Device` from ironlib and use `common.Device` directly.

File: model/device.go (hunk 1 above)
Comment (joelrebel) [reply model/device.go:23]: The `ironlib.Device` object is being extended to include `OemComponents` https://github.com/metal-toolbox/ironlib/blob/61b187384a6779ad0141df6861cffebe4051d238/model/device.go#L27

Comment (diogomatsubara) [reply model/device.go:23]: ah, I clearly missed that.

**Review by diogomatsubara — APPROVED** [review]

**Pull request merged**